File: apps/myjobhunter/backend/app/schemas/platform/invite_status.py
"""Computed-state enum for an invite row.

The DB doesn't store this directly — the row's accepted_at / expires_at
columns are the source of truth. This enum is the API-surface
projection: every row is exactly one of these, every time, derived by
``InviteService.compute_status``.
"""
from __future__ import annotations

from enum import Enum


class InviteStatus(str, Enum):
    PENDING = "pending"
    ACCEPTED = "accepted"
    EXPIRED = "expired"

File: apps/myjobhunter/frontend/src/types/invite/invite-status.ts
/**
 * Computed-state enum for a platform invite. Mirrors
 * apps/myjobhunter/backend/app/schemas/platform/invite_status.py exactly.
 *
 * The backend recomputes status on every read from `accepted_at` and
 * `expires_at`; the frontend never derives it locally — always reads
 * the API value.
 */
export const INVITE_STATUS = {
  PENDING: "pending",
  ACCEPTED: "accepted",
  EXPIRED: "expired",
} as const;

export type InviteStatus = (typeof INVITE_STATUS)[keyof typeof INVITE_STATUS];
